consensus.go

```diff
@@ -273,7 +286,9 @@ func (c *ConsensusCmd) RunNode() {
 
 					go func() {
 						c.mockExecution(slotLog, block)
-						ForkchoiceUpdated(c.ctx, c.engine, c.log, Bytes32(block.Hash()), Bytes32(block.Hash()))
+						latest := Bytes32(block.Hash())
+						final := Bytes32(finalizedHash)
+						ForkchoiceUpdated(c.ctx, c.engine, c.log, latest, latest, final, nil)
 					}()
```

Comment:
We should pass the block and finalizedHash as arguments into the goroutine, to avoid concurrency problems with the surrounding loop.